### ISSUE TITLE
Background daemon logging activity

### ISSUE THREAD

**Issue by luke-jr** [raboof/nethogs#127]
It would be nice for a daemon that can just run in the background constantly keeping track of usage by processes for later analysis, similar to Android's data usage display.

This would ideally mean logging the interface and associated wireless access point (so one can differentiate between mobile data and home wifi).

**Comment by raboof** [comment]
'Background daemon logging' would be a cool feature for nethogs indeed.

If 'just' want to collect information on the interface and associated wireless AP's, however, nethogs might be overkill: nethogs is typically only interesting if you also want to see/keep associated process information.

**Comment by luke-jr** [comment]
No, in fact I want to see what programs use how much data, while I am connected to cellular networks. So if at the end of a day, I see KMail (for example) using too much, I will script my device to SIGSTOP it while connected to cellular. :)

**Comment by raboof** [comment]
Aah - then indeed nethogs makes sense, but currently it was designed for 'interactive' use. I don't have plans to make it suitable for 'daemon' use right now, but it'd be a cool thing to have :).

**Comment by luke-jr** [comment]
How about adding the extra info (interface at least) to trace mode, and accepting SIGUSR1 to ask it to immediately print and move on to the next chunk? Then I can have an external tool that watches just the wifi info, and sends SIGUSR1 when the network changes..

**Comment by raboof** [comment]
I'd certainly be in favor of triggering a tracemode refresh on SIGUSR1, PR welcome. Question is whether we should clear buffers/averages at that point, I don't think so.

Not sure adding interface would be very useful, usually you're probably just monitoring one interface anyway?

**Comment by luke-jr** [comment]
It could be useful in case there are multiple network connections. Although I am not sure if nethogs splits up the activity right now?

**Comment by luke-jr** [comment]
(eg, wifi + wired both active simultaneously)

**Comment by raboof** [comment]
I haven't checked but I don't think so - you could easily start up as many nethogs processes as you have interfaces to monitor, though.
